src/Fmt_ast.ml

```diff
@@ -764,8 +764,12 @@ and fmt_core_type c ?(box = true) ?(in_type_declaration = false) ?pro
         $ fmt ".@ "
         $ fmt_core_type c ~box:false (sub_typ ~ctx t) )
   | Ptyp_tuple typs ->
+      let wrap_if =
+        if Poly.(c.conf.braces_space = `Loose) then wrap_fits_breaks_if
+        else wrap_if_str
+      in
```

Comment:
There is an unfortunate amount of duplication at the moment. Since the objective of this PR seems to be essentially to defeat `wrap_fits_breaks`, it seems to me that it would be clearer and easier to adjust it directly instead. What about passing `c` to `wrap_fits_breaks` instead? I think this boils down to whether there are cases there even with this added option set, you still want the current behavior of `wrap_fits_breaks`? I don't conceptually understand what is different about the cases that add inner spaces with this PR compared to those that have been changed.

Reply:
I factorized the code, but restricted the option for parens, because the formatting is not looking good when there are semicolons involved for example.